pkg/network/ovn_kubernetes.go

```diff
@@ -668,10 +668,8 @@ func shouldRenderIPsec(conf *operv1.OVNKubernetesConfig, bootstrapResult *bootst
 	// Wait for MCO to be ready unless we had already rendered the IPsec MachineConfig.
 	renderCNOIPsecMachineConfig = renderCNOIPsecMachineConfig && (isCNOIPsecMachineConfigPresent || isMachineConfigClusterOperatorReady)
 
-	// We render OVN IPsec if East-West IPsec is enabled or it's upgrade is in progress.
-	// If NS IPsec is enabled as well, we need to wait to IPsec MachineConfig
-	// to be active if it's not an upgrade and not a hypershift hosted cluster.
-	renderIPsecOVN = (renderIPsecHostDaemonSet || renderIPsecContainerizedDaemonSet) && mode == operv1.IPsecModeFull
+	// We render OVN IPsec as long as East-West IPsec is enabled.
+	renderIPsecOVN = mode == operv1.IPsecModeFull
```

Comment:
Why not
```
renderIPsecOVN = (renderIPsecHostDaemonSet || renderIPsecContainerizedDaemonSet || isOVNIPsecActive) && mode == operv1.IPsecModeFull
```

Why don't we need to wait anymore for the NS IPSec to be deployed before activating OVN IPSec?

Also, it's best if the comment explains why we are doing what we are doing, not just what we are doing.

Reply:
>Why don't we need to wait anymore for the NS IPSec to be deployed before activating OVN IPSec?

activating OVN IPSec at this case won't trigger anything because ovs-monitor-ipsec script is not started yet. but still your suggestion on checking `isOVNIPsecActive` makes more sense as it enables IPsec OVN only when needed. changed it. added some info about explaining 'why', hope that's ok now.

Reply:
Thanks @jcaamano for your time and patience, it was a great long discussion to update IPsec state machine with up to  date information. Updated the PR as we discussed.